app/client/templates/elements/eventsTable.js

```diff
@@ -49,7 +49,7 @@ Template['elements_event_table'].helpers({
             searchQuery = TemplateVar.get('search'),
             limit = TemplateVar.get('limit'),
             collection = Events,
-            selector = this.ids ? {_id: {$in: this.ids.slice(Number((limit+1)*-1))}} : {}; 
+            selector = this.ids ? {_id: {$in: this.ids.slice(Number((limit+50)*-1))}} : {}; 
```

Comment:
The ids. come from the contracts `transactions` and `contractEvent` properties. These get filled once an event comes in. Means new events/tx get added to the end of the array. So your solution should work fine. Except when events are coming out of order, e.g. when missing events are fetched, this could only be complicated on the last events shown in the list, the first ones should be fine.

We just have to keep that in mind.